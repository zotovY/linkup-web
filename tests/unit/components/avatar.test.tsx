/// <reference types="@types/jest" />
import { render } from "../../test-utils";
import '@testing-library/jest-dom';
import 'jest-styled-components'

import AvatarComponent, { Props } from "@/components/avatar";
import { cleanup } from "@testing-library/react";


describe("Testing avatar component", () => {

    afterEach(cleanup);

    const props: Props = {
        profileImagePath: "https://avatars.githubusercontent.com/u/35657233?v=4"
    };


    test("should renders correctly", () => {
        const { asFragment } = render(<AvatarComponent { ...props }/>);
        expect(asFragment()).toMatchSnapshot();
    });

    test("should render placeholder", () => {
        const { getByTestId } = render(<AvatarComponent/>);
        expect(getByTestId("avatar")).toHaveStyle("background-image: url(\"/images/user-placeholder.png\");");
    });
});
