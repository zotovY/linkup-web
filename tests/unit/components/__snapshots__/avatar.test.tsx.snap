// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing avatar component should renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  width: 90px;
  height: 90px;
  border-radius: 50%;
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  background-color: #D9DDE9;
}

<div
    class="c0 user-avatar"
    data-testid="avatar"
    style="background-image: url(https://avatars.githubusercontent.com/u/35657233?v=4);"
  />
</DocumentFragment>
`;
